### PR TITLE
Make AccessIntent an IntFlag and add READ, WRITE

### PR DESCRIPTION
**Pull request by jdahm**
## Description

Minor change to allow for write checking independent from reads within the classic toolchain.

For now it is difficult to propagate similar changes to GTC, since gtir, which `make_args_from_gtir` uses, does not make it easy to separate reads from writes.

### CODE REVIEW

File: src/gt4py/ir/nodes.py [hunk 1]
```diff
@@ -287,9 +287,10 @@ def __str__(self):
 
 
 @enum.unique
-class AccessIntent(enum.Enum):
-    READ_ONLY = 0
-    READ_WRITE = 1
+class AccessIntent(enum.IntFlag):
+    READ = 1
+    WRITE = 2
+    READ_WRITE = 3
```
Comment (egparedes) [inline src/gt4py/ir/nodes.py:293]: Isn't `READ_WRITE` redundant here? I mean, if `AccessIntent` is a `Flag`, it should be possible to set both `READ` and `WRITE` at the same time, or did I get it wrong?

**Review by egparedes — COMMENTED** [review]
It looks good. I only have a question about the implementation using `IntFlag`s.

**Comment by egparedes** [comment]
Additionally, it looks like there are now conflicts that should be addressed.

File: src/gt4py/ir/nodes.py (hunk 1 above)
Comment (jdahm) [reply src/gt4py/ir/nodes.py:293]: It's so that it can be specified as a shortcut for `AccessIntent.READ & AccessIntent.WRITE`.

Comment (jdahm) [reply src/gt4py/ir/nodes.py:293]: We could instead use `READ_WRITE = READ | WRITE`

**Review by egparedes — APPROVED** [review]
Approved.

**Pull request merged**